packages/admin-ui/src/filters/filter-multiple.state.tsx

```diff
@@ -1,11 +1,82 @@
-import type {
-  UseFilterMultipleReturn,
-  UseFilterMultipleStateProps,
-} from './base-filter.state'
-import { useBaseFilterState } from './base-filter.state'
-
-export function useFilterMultipleState(
-  props: UseFilterMultipleStateProps
-): UseFilterMultipleReturn {
-  return useBaseFilterState({ ...props, selectionMode: 'multiple' })
+import { useState, useCallback, useEffect } from 'react'
+
+import type { ComboboxMultipleState } from '../combobox'
+import { useComboboxMultipleState } from '../combobox'
+import { useMenuState } from 'ariakit/menu'
+import type { GenericFilterStateReturn, FilterItem } from './filter.state'
+
+export function useFilterMultipleState<T extends FilterItem>(
+  props: UseFilterMultipleStateProps<T>
+): UseFilterMultipleReturn<T> {
+  const { items, label, initialApplied, onChange = () => {} } = props
+
+  const [appliedItems, setAppliedItems] = useState<T[]>([])
+
+  const comboboxMultiple = useComboboxMultipleState<T>({
+    getOptionValue: (op) => op.id,
```

Comment:
**Suggestion**: I really prefer let the params name more explicit for code readability reasons + clean code. 
```suggestion
    getOptionValue: (option) => option.id,
```